pkgs/by-name/af/affine/package.nix

```
@@ -35,212 +36,203 @@ let
+  version = "0.19.6";
+  src = fetchFromGitHub {
+    owner = "toeverything";
+    repo = "AFFiNE";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-BydTNE36oRIxr2lTnc2+EY0lvMXn4NTLB4EjqzhdjGk=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    src = finalAttrs.src;
+    hash = "sha256-racjpf0VgNod6OxWKSaCbKS9fEkInpDyhVbAHfYWIDo=";
+  };
+  yarnOfflineCache = stdenvNoCC.mkDerivation {
+    name = "yarn-offline-cache";
+    src = finalAttrs.src;
+    nativeBuildInputs = [
+      yarn
+      cacert
+      writableTmpDirAsHomeHook
+    ];
+    supportedArchitectures = builtins.toJSON {
+      os = [
+        "darwin"
+        "linux"
+      ];
+      cpu = [
+        "arm64"
+        "x64"
+      ];
+      libc = [
+        "glibc"
+        "musl"
+      ];
+    };
+    env.CI = "1";
+    buildPhase = ''
+      runHook preBuild
+
+      mkdir -p $out
+      yarn config set enableTelemetry false
+      yarn config set cacheFolder $out
+      yarn config set enableGlobalCache false
+      yarn config set supportedArchitectures --json "$supportedArchitectures"
+
+      yarn install --immutable --mode=skip-build
+
+      runHook postBuild
+    '';
+    dontInstall = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-E9l5zjOOfyDBzYJOU94VrRvt7Hi4XkRTDav9bVlXvlQ=";
+  };
+  nativeBuildInputs =
+    [
+      nodejs
+      yarn
+      cargo
+      rustc
+      findutils
+      zip
+      jq
+      rsync
+      writableTmpDirAsHomeHook
+    ]
+    ++ lib.optionals hostPlatform.isLinux [
+      copyDesktopItems
+      makeWrapper
+    ];
+
+  env.CI = "1";
+
+  # Remove code under The AFFiNE Enterprise Edition (EE) license.
+  # Keep file package.json for `yarn install --immutable` lockfile check.
+  postPatch = ''
+    BACKEND_SERVER_PACKAGE_JSON="$(jq 'del(.scripts.postinstall)' packages/backend/server/package.json)"
+    rm -rf packages/backend/server/{.*,*}
+    echo "$BACKEND_SERVER_PACKAGE_JSON" > packages/backend/server/package.json
+  '';
+
+  configurePhase = ''
+    runHook preConfigurePhase
+
+    # cargo config
+    mkdir -p .cargo
+    cat $cargoDeps/.cargo/config.toml >> .cargo/config.toml
+    ln -s $cargoDeps @vendor@
+
+    # yarn config
+    yarn config set enableTelemetry false
+    yarn config set enableGlobalCache false
+    yarn config set cacheFolder $yarnOfflineCache
+
+    # electron config
+    ELECTRON_VERSION_IN_LOCKFILE=$(yarn why electron --json | tail --lines 1 | jq --raw-output '.children | to_entries | first | .key ' | cut -d : -f 2)
+    rsync --archive --chmod=u+w "${electron.dist}/" $HOME/.electron-prebuilt-zip-tmp
+    export ELECTRON_FORGE_ELECTRON_ZIP_DIR=$PWD/.electron_zip_dir
+    mkdir -p $ELECTRON_FORGE_ELECTRON_ZIP_DIR
+    (cd $HOME/.electron-prebuilt-zip-tmp && zip --recurse-paths - .) > $ELECTRON_FORGE_ELECTRON_ZIP_DIR/electron-v$ELECTRON_VERSION_IN_LOCKFILE-${nodePlatform}-${nodeArch}.zip
+    export ELECTRON_SKIP_BINARY_DOWNLOAD=1
+
+    runHook postConfigurePhase
+  '';
```

```suggestion
  '';

```
that new line improves readability a lot